[testplan] Macsec delete add in use policy

Azarack opened the pull request:
<!--
Please make sure you've read and understood our contributing guidelines;
https://github.com/sonic-net/SONiC/blob/gh-pages/CONTRIBUTING.md

Please provide following information to help code review process a bit easier:
-->
### Description of PR
<!--
- Please include a summary of the change and which issue is fixed.
- Please also include relevant motivation and context. Where should reviewer start? background context?
- List any dependencies that are required for this change.
-->

Summary:
Fixes # (issue)
Test case to verify delete and add behavior on Sonic.

### Type of change

<!--
- Fill x for your type of change.
- e.g.
- [x] Bug fix
-->

- [ ] Bug fix
- [ ] Testbed and Framework(new/improvement)
- [X] Test case(new/improvement)


### Back port request
- [ ] 201911
- [ ] 202012
- [ ] 202205
- [ ] 202305
- [ ] 202311

### Approach
#### What is the motivation for this PR?

#### How did you do it?

#### How did you verify/test it?
Virtual T2 testbed

#### Any platform specific information?
Only tested on T2

#### Supported testbed topology if it's a new test case?
T2

### Documentation
<!--
(If it's a new feature, new test case)
Did you update documentation/Wiki relevant to your implementation?
Link to the wiki page?
-->

yxieca commented:
@Pterosaur @rlhui can you assign someone to review this test plan?

judyjoseph commented:
> @Pterosaur @rlhui can you assign someone to review this test plan?

I will take a look @yxieca

judyjoseph commented on tests/macsec/test_delete_add.py:
```diff
@@ -0,0 +1,43 @@
+]
+
+
+def test_delete_add_policy(macsec_duthost, startup_macsec, shutdown_macsec, ctrl_links, profile_name, default_priority,
```
When this test is called, the assumption is that "profile_name" is already used and referenced in some interface? Can we check that

Azarack replied:
I can add a check, but the test is labeled as macsec required.  Is there a scenario where a macsec test somehow runs on a setup that doesn't have macsec enabled on it?

judyjoseph commented:
Please add a test run results ( for single and multi-asic devices/linecards) in PR description -- that will be helpful

Azarack commented on tests/macsec/test_delete_add.py:
```diff
@@ -0,0 +1,43 @@
+                           cipher_suite, primary_cak, primary_ckn, policy, send_sci, capsys, rekey_period):
+    port_name, nbr = list(ctrl_links.items())[0]
+    if macsec_duthost.is_multi_asic:
+        ns = " -n " + macsec_duthost.get_port_asic_instance(port_name).asic_index
```
Yes, I have corrected this.

Azarack commented:
> Please add a test run results ( for single and multi-asic devices/linecards) in PR description -- that will be helpful

I'm not sure what exactly you are asking for here, but I don't have access to multi-asic devices for testing unfortunately.

Azarack commented:
@judyjoseph Can you review the comments I added, please?

judyjoseph commented:
Please could you update the testcase run/pass results in the PR description, thanks

Azarack commented:
> Please could you update the testcase run/pass results in the PR description, thanks

@judyjoseph Are you asking for this?

1 passed, 1 warning in 337.14s (0:05:37)

Azarack commented:
@judyjoseph can you confirm what information you are asking for, please?

judyjoseph commented:
Thanks @Azarack  for addressing comments and sorry for delay from my end.

Could you add this new test in the same file : [test_deployment.py](https://github.com/sonic-net/sonic-mgmt/blob/7789f24847943a81f95f3d190818e2da3d8db1fb/tests/macsec/test_deployment.py). As this is a deployment usecase, better to segregate all tests under this umbrella

Azarack commented:
> Thanks @Azarack for addressing comments and sorry for delay from my end.
> 
> Could you add this new test in the same file : [test_deployment.py](https://github.com/sonic-net/sonic-mgmt/blob/7789f24847943a81f95f3d190818e2da3d8db1fb/tests/macsec/test_deployment.py). As this is a deployment usecase, better to segregate all tests under this umbrella

@judyjoseph I have moved the test case into the deployment file.

judyjoseph commented on tests/macsec/test_deployment.py:
```diff
@@ -8,7 +9,7 @@
 pytestmark = [
     pytest.mark.macsec_required,
-    pytest.mark.topology("t0", "t2"),
+    pytest.mark.topology("t2"),
```
Hello @Azarack we cannot remove t0 here as we still run the macsec tests for PR sanity with t0 toplogy

Azarack replied:
I added it back.  Can you check the pipeline as everything is failing on the T0 run with no real error message.  Both this PR and others I have recently submitted.

judyjoseph commented:
@Azarack  please add a test run snapshot -- if have got one in hand.

judyjoseph commented:
/azp run

judyjoseph commented:
/azp run

Azarack commented:
> @Azarack please add a test run snapshot -- if have got one in hand.

I merged the main branch in to fix the conflict in the testplan doc, but when I try to run I get this error:
ERROR:pytest_ansible.units:No galaxy.yml file found, plugin not activated

I am not sure what this means or how to resolve.

judyjoseph commented:
@Azarack you have taken in more commits into this PR. I think the merge earlier was not good - it has 966 files modified

Azarack commented:
> @Azarack you have taken in more commits into this PR. I think the merge earlier was not good - it has 966 files modified

I brought in (or tried to) the latest updates to the master branch.  I used the web ui to sync my fork to the upstream's branch and merged this in.  Is there a different way I should be doing this task?

Azarack commented:
@judyjoseph I have tried to revert to previous commits, but the extra files are still there.  From what I can understand there may have been a merge on the master branch that is now causing this.

judyjoseph commented:
> @judyjoseph I have tried to revert to previous commits, but the extra files are still there. From what I can understand there may have been a merge on the master branch that is now causing this.

@Azarack please create a new PR if you are not able to remove the commits. I feel that is easier

Azarack commented:
This change has been resubmitted through https://github.com/sonic-net/sonic-mgmt/pull/15961
